src/main/java/io/r2dbc/postgresql/codec/Point.java

```diff
@@ -0,0 +1,118 @@
+package io.r2dbc.postgresql.codec;
+
+/**
+ * <p>It maps to the point datatype in org.postgresql.</p>
+ *
+ * <p>This implements a version of java.awt.Point, except it uses double to represent the coordinates.</p>
+ */
+public class Point {
+    private double x;
+    private double y;
+
+    /**
+     * @param x coordinate
+     * @param y coordinate
+     */
+    public Point(double x, double y) {
+        this.x = x;
+        this.y = y;
+    }
+
+    public double getX() {
+        return x;
+    }
+
+    public void setX(double x) {
+        this.x = x;
+    }
+
+    public double getY() {
+        return y;
+    }
+
+    public void setY(double y) {
+        this.y = y;
+    }
+
+    /**
+     * Translate the point by the supplied amount.
+     *
+     * @param x integer amount to add on the x axis
+     * @param y integer amount to add on the y axis
+     */
+    public void translate(int x, int y) {
+        translate((double) x, (double) y);
+    }
+
+    /**
+     * Translate the point by the supplied amount.
+     *
+     * @param x double amount to add on the x axis
+     * @param y double amount to add on the y axis
+     */
+    public void translate(double x, double y) {
+        this.x += x;
+        this.y += y;
+    }
+
+    /**
+     * Moves the point to the supplied coordinates.
+     *
+     * @param x integer coordinate
+     * @param y integer coordinate
+     */
+    public void move(int x, int y) {
+        setLocation(x, y);
+    }
+
+    /**
+     * Moves the point to the supplied coordinates.
+     *
+     * @param x double coordinate
+     * @param y double coordinate
+     */
+    public void move(double x, double y) {
+        this.x = x;
+        this.y = y;
+    }
+
+    /**
+     * Moves the point to the supplied coordinates. refer to java.awt.Point for description of this.
+     *
+     * @param x integer coordinate
+     * @param y integer coordinate
+     * @see java.awt.Point
+     */
+    public void setLocation(int x, int y) {
+        move((double) x, (double) y);
+    }
+
+    /**
+     * Moves the point to the supplied java.awt.Point refer to java.awt.Point for description of this.
+     *
+     * @param p Point to move to
+     * @see java.awt.Point
+     */
+    public void setLocation(java.awt.Point p) {
```

Comment:
We should not use AWT at all.

Reply:
yes, I was thinking the same. I'll remove this.